urdf2webots/writeRobot.py

```diff
@@ -129,7 +129,16 @@ def URDFLink(robotFile, link, level, parentList,
                 if not haveChild:
                     haveChild = True
                     robotFile.write((level + 1) * indent + 'children [\n')
-                sensor.export(robotFile, level + 2)
+                if hasattr(sensor, 'isImager') and sensor.isImager:
+                    robotFile.write((level + 2) * indent + 'Transform {\n')
+                    robotFile.write((level + 3) * indent + 'translation 0 0 0\n')
+                    robotFile.write((level + 3) * indent + 'rotation 0.577350 -0.577350 0.577350 2.094395\n')
+                    robotFile.write((level + 3) * indent + 'children [\n')
+                    sensor.export(robotFile, level + 4)
+                    robotFile.write((level + 3) * indent + ']\n')
+                    robotFile.write((level + 2) * indent + '}\n')
```

Comment:
I agree on the rotation need, but it would be cleaner to combine it with the precedent rotation rather than adding a transform node in another transform node.

https://github.com/cyberbotics/urdf2webots/blob/3122b3164fd38519e00e9fbc2172fbb02751b836/urdf2webots/writeRobot.py#L120-L123

Reply:
Yes, I agree. That was something, I've been looked out in the first place but seems to require many changes.
Doing it this way (as extra transform) does not interfere with the existing frames, which may impact how meshes/shapes are visualizes. Imagine very simple models, which do not consider such optical frames.

Reply:
I think you just have to use the `combineRotations(rotation1, rotation2)` function to combine the rotations and you do it only for the cameras.

I do not see the reason why this would have an impact on the meshes and shapes, the camera is a sensor contained in a link without shapes or meshes and then will be joint to the link containing the shapes and meshes no?

Reply:
> I think you just have to use the `combineRotations(rotation1, rotation2)` function to combine the rotations and you do it only for the cameras.

Ok, sounds good!

> I do not see the reason why this would have an impact on the meshes and shapes, the camera is a sensor contained in a link without shapes or meshes and then will be joint to the link containing the shapes and meshes no?

I fully agree with you that this is the proper modelling approach. My concern was only that it may brake some poor designed URDFs, which use the same link for the mesh and the camera.

Reply:
> I fully agree with you that this is the proper modelling approach. My concern only was that it may brake some poorly designed URDFs that use the same link for the mesh and the camera.

Well spotted, with this case we prefer to keep the transform only on the device.

I let you change the targeted branch to `master` and then this PR will be fine!

Reply:
Switched target branch to `master`